Dockerfile

```diff
@@ -11,8 +14,13 @@ RUN conda install -c conda-forge google-cloud-sdk && \
         git gcc rsync sudo patch openssh-server \
         pciutils nano fuse socat netcat-openbsd curl rsync vim tini && \
     rm -rf /var/lib/apt/lists/* && \
-    # Install kubectl
-    curl -LO "https://dl.k8s.io/release/v1.31.6/bin/linux/amd64/kubectl" && \
+    # Install kubectl based on architecture
+    ARCH=${TARGETARCH:-$(case "$(uname -m)" in \
```

Comment:
Should we also update the kubectl installation when user runs a controller on arm machines:

https://github.com/skypilot-org/skypilot/blob/65ac50ea3a31a7c5efacb807392208028730b54d/sky/utils/controller_utils.py#L248-L252

Reply:
Good catch! Updated